pandas/core/reshape/tile.py

```diff
@@ -413,7 +433,10 @@ def _bins_to_cuts(
                 )
 
         if not is_categorical_dtype(labels):
-            labels = Categorical(labels, categories=labels, ordered=True)
+            if len(set(labels)) == len(labels):
```

Comment:
why do you need this check here?

Reply:
We need to differentiate between the case of duplicated labels or not, in order to maintain the original functionality.

Categories cannot be passed when duplicates since it will raise an error about categories needing to be unique.

Reply:
I think we want to be a bit more strict here and raise if `ordered=True` and labels are not unique.  The current behavior passes this case on to the `else` clause which produces invalid output:
```python
In [1]: import pandas as pd; pd.__version__
Out[1]: '1.1.0.dev0+1388.g5882374c3'

In [2]: pd.cut(range(10), bins=4, labels=list('cbaa'), ordered=True)
Out[2]: 
[c, c, c, b, b, a, a, a, a, a]
Categories (3, object): [a < b < c]
```
Note that the labels are not in the user specified order (`a < b < c` but `c` was specified first).

Perhaps something like:
```python
if ordered and len(set(labels)) != len(labels):
    raise ValueError("labels must be unique if ordered=True; pass ordered=False for duplicate labels")

# specify categories to maintain label ordering if ordered=True otherwise infer to handle dupes
categories = labels if ordered else None
labels = Categorical(labels, categories=categories, ordered=ordered)
```
Should probably do the check with the other input validation at the beginning of `_bins_to_cuts`.  Could maybe combine the two lines that create the `Categorical` into a single line.